pilot/pkg/bootstrap/server.go

```diff
@@ -836,10 +815,112 @@ func (s *Server) initSecureGrpcListener(args *PilotArgs) error {
 	}
 
 	// run secure grpc server for Istiod - using DNS-based certs from K8S
-	err = s.initSecureGrpcServer(port, args.KeepaliveOptions, args.TLSOptions)
-	if err != nil {
+	if err = s.initSecureGrpcServer(port, args.KeepaliveOptions, args.TLSOptions); err != nil {
 		return err
 	}
 
 	return nil
 }
+
+// initCertificateWatches sets up  watches for the certs.
```

Comment:
Just a reminder: in the past we had problems with pods running on nodes or apps where a lot of watchers were used. There are limits on the kernel - and it leads to hard to debug problems and is not easy to test. 

Would be great to double check this falls back gracefully and doesn't crash if watching files doesn't work.